### PR TITLE
[PyOV] Extend Python API with ROIAlignRotated-15 

### PR DESCRIPTION
**Pull request by hub-bla**
### Details:
 - Add Python API for `ROIAlignRotated-15`

### Tickets:
 - 141569

### CODE REVIEW

File: src/bindings/python/src/openvino/runtime/opset15/ops.py [hunk 1]
```diff
@@ -144,3 +144,45 @@ def embedding_bag_packed(
...
+@nameable_op
+def roi_align_rotated(
+    data: NodeInput,
+    rois: NodeInput,
+    batch_indices: NodeInput,
+    pooled_h: int,
+    pooled_w: int,
+    sampling_ratio: int,
+    spatial_scale: float,
+    clockwise_mode: bool,
```
Comment (mitruska) [inline src/bindings/python/src/openvino/runtime/opset15/ops.py:158]: (Side comment, it's not a change request:)
The `clockwise_mode` could be an optional attribute, but
it is actually required by ROIAlignRotated C++ constructor. 
So it's okay to keep it as is, aligned with C++ API.

File: src/bindings/python/tests/test_graph/test_create_op.py [hunk 2]
```diff
@@ -885,6 +886,37 @@ def test_roi_align(data_shape, rois, batch_indices, pooled_h, pooled_w, sampling
     assert list(node.get_output_shape(0)) == expected_shape
 
 
+@pytest.mark.parametrize(
+    ("data_shape", "rois", "batch_indices", "pooled_h", "pooled_w", "sampling_ratio", "spatial_scale", "clockwise_mode", "expected_shape"),
```
Comment (mitruska) [inline src/bindings/python/tests/test_graph/test_create_op.py:890]: (Detail, optional) The test is parametrized by the input shapes, that could be reflected in the test param names for a better readability.
```suggestion
    ("data_shape", "rois_shape", "batch_indices_shape", "pooled_h", "pooled_w", "sampling_ratio", "spatial_scale", "clockwise_mode", "expected_shape"),
```

**Review by mitruska — APPROVED** [review]
Thank you for your contribution 🚀 
This PR LGTM.

File: src/bindings/python/tests/test_graph/test_create_op.py (hunk 2 above)
Comment (hub-bla) [reply src/bindings/python/tests/test_graph/test_create_op.py:890]: `test_roi_align` has similar arguments. Should I also modify those as well?
https://github.com/openvinotoolkit/openvino/blob/b0b36a6c28b4046f817025f427b24b6f30848951/src/bindings/python/tests/test_graph/test_create_op.py#L856-L857

**Review by p-wysocki — APPROVED** [review]
Looks great, thank you for your contribution!

**Comment by hub-bla** [comment]
Thank you for your dedication in developing and maintaining. I'm genuinely impressed by how well everything is organized :)

**Comment by mlukasze** [comment]
build_jenkins

**Comment by mlukasze** [comment]
build_jenkins

File: src/bindings/python/tests/test_graph/test_create_op.py (hunk 2 above)
Comment (mitruska) [reply src/bindings/python/tests/test_graph/test_create_op.py:890]: Seems like the naming could be slightly adjusted for both tests, but the PR is already good.
Let's wait for the green CI (validation cycle) and we will merge it (no more changes required).

**Comment by hub-bla** [comment]
Hey @mlukasze, I noticed that there's a failed Jenkins check. Could you provide me with some hints on how to fix this issue? Thanks!

**Comment by mitruska** [comment]
> Could you provide me with some hints on how to fix this issue? Thanks!

@hub-bla That looks like an infrastructure issue, update to the master branch and CI rebuild should help (already done).
It's on our side, further action is not required so far:) Thanks!

**Comment by mlukasze** [comment]
build_jenkins

**Comment by hub-bla** [comment]
@mitruska It looks like the CI process has stalled on two ubuntu checks. I took a quick glance at them yesterday, and there haven’t been any changes since then.

**Comment by mitruska** [comment]
build_jenkins

**Pull request merged**